src/components/Input.react.js

```diff
@@ -89,6 +92,13 @@ export default class Input extends PureComponent {
         );
     }
 
+    getValueAsNumber(value) {
+        if (parseInt(value, 10) !== parseFloat(value)) {
+            return parseFloat(value);
+        }
+        return parseInt(value, 10);
```

Comment:
Why not save the result of `parseInt` and `parseFloat` to variables and test / return those instead of re-parsing?

Reply:
Also, I think this will behave differently from the current implementation for something like "1e4", with `current.valueAsNumber` returning `10000` and `parseInt` returning `1`.

When possible we've been using https://github.com/plotly/fast-isnumeric combined with `+myString` to validate and cast strings into numbers.